src/dfx-core/assets/build.rs

```diff
@@ -0,0 +1,54 @@
+use std::{env, fs::File, io::Write, path::Path};
+
+use itertools::Itertools;
+use serde_json::Value;
+
+fn define_well_known_canisters() {
+    let well_known_canisters = serde_json::from_str::<Value>(
+        &std::fs::read_to_string(format!(
+            "{}/assets/canister_ids.json",
+            env!("CARGO_MANIFEST_DIR")
+        ))
+        .unwrap(),
+    )
+    .unwrap();
+    let well_known_canisters = well_known_canisters.as_object().unwrap();
+    let well_known_canisters = well_known_canisters.iter().map(|(key, val)| {
+        (
+            key.as_str(),
+            val.as_object()
+                .unwrap()
+                .values()
+                .last()
```

Comment:
The principals are all the same for all networks in the included canister_ids.json, but what would you think about looking up entry for `mainnet` here instead of whatever's last?